packages/babel-check-binary-expressions/src/index.js

```diff
@@ -1,6 +1,24 @@
-export default function(left, right) {
-  // eslint-disable-next-line no-undef
-  if (left instanceof BigInt && right instanceof BigInt) {
+import isNumber from "lodash/isNumber";
+import isString from "lodash/isString";
+
+function isBigInt(x) {
+  return x instanceof global.BigInt;
+}
+
+export default function(left, right, operator) {
+  // TODO: don't throw for operations that can be performed on BigInt and Number (i.e. comparisons)
+  if (
+    (isBigInt(left) && isNumber(right)) ||
+    (isNumber(left) && isBigInt(right))
+  ) {
+    throw new TypeError(`Cannot perform ${operator} on a BigInt and a Number`);
+  }
+
+  if (isString(left) || isString(right)) {
+    return left.toString() + right.toString();
+  }
+
+  if (isBigInt(left) && isBigInt(right)) {
     return left.plus(right);
```

Comment:
Better to not use a method called `plus` but instead a function. The problem with a method is that it's easily accessible to users (who will be missing it in the real version). It's better to use a function, which we could refrain from exporting to them.